src/domains/ValuesDomain.js

```diff
@@ -259,6 +259,52 @@ export default class ValuesDomain {
     invariant(false, "unimplemented " + op);
   }
 
+  static logicalOp(realm: Realm, op: BabelNodeLogicalOperator, left: ValuesDomain, right: ValuesDomain): ValuesDomain {
+    let leftElements = left._elements;
+    let rightElements = right._elements;
+    // Return top if left and/or right are top or if the size of the value set would get to be quite large.
+    // Note: the larger the set of values, the less we know and therefore the less we get value from computing
+    // all of these values. TODO: probably the upper bound can be quite a bit smaller.
+    if (!leftElements || !rightElements || leftElements.size > 100 || rightElements.size > 100)
+      return ValuesDomain.topVal;
+    let resultSet = new Set();
+    let savedHandler = realm.errorHandler;
+    let savedIsReadOnly = realm.isReadOnly;
```

Comment:
You save and restore isReadOnly, but I don't see where it would get mutated?

Reply:
Is this a bigger issue that also applied to other existing code, or am I missing something?

Reply:
It needs to get mutated here.